RSSReader/View/WebViewController.m

```diff
@@ -0,0 +1,172 @@
+//
+//  WebViewController.m
+//  RSSReader
+//
+//  Created by Uladzislau Stankevich on 7.01.21.
+//
+
+#import "WebViewController.h"
+#import <WebKit/WebKit.h>
+
+@interface WebViewController () <UIScrollViewDelegate>
+
+@property (nonatomic, assign)WKWebView *webView;
+@property (nonatomic, retain)UIBarButtonItem *backBarButtonItem;
+@property (nonatomic, retain)UIBarButtonItem *forwardBarButtonItem;
+@property (nonatomic, retain)UIBarButtonItem *refreshBarButtonItem;
+@property (nonatomic, retain)UIBarButtonItem *stopBarButtonItem;
+@property (nonatomic, retain)UIBarButtonItem *safariBarButtonItem;
+@property (nonatomic, retain)UIBarButtonItem *spacerBarButtonItem;
+@property (nonatomic) CGFloat contentOffsetLastY;
+
+@end
+
+@implementation WebViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    self.view.backgroundColor = UIColor.whiteColor;
+    [self setupLayout];
+    [self setupToolBar];
+    [self openURL];
+}
+
+- (instancetype)initWithURL:(NSURL *)url {
+    self = [super init];
+    if (self) {
+        _url = [url retain];
+    }
+    return self;
+}
+
+- (void)setupLayout {
+    [self.view addSubview:self.webView];
+    
+    [NSLayoutConstraint activateConstraints:@[
+        [self.webView.leadingAnchor constraintEqualToAnchor:self.view.leadingAnchor],
+        [self.webView.topAnchor constraintEqualToAnchor:self.view.safeAreaLayoutGuide.topAnchor],
+        [self.webView.trailingAnchor constraintEqualToAnchor:self.view.trailingAnchor],
+        [self.webView.bottomAnchor constraintEqualToAnchor:self.view.bottomAnchor],
+    ]];
+}
+
+- (WKWebView *)webView {
+    if (!_webView) {
+        _webView = [WKWebView new];
+        _webView.translatesAutoresizingMaskIntoConstraints = NO;
+        _webView.scrollView.delegate = self;
+        _webView.allowsBackForwardNavigationGestures = true;
+    }
+    return _webView;
+}
+
+- (void)openURL {
+    NSURLRequest *nsrequest=[NSURLRequest requestWithURL:self.url];
```

Comment:
there is no sense in creating another local variable. Just
[self.webView loadRequest:[NSURLRequest requestWithURL:self.url]];

Reply:
fixed